### PR TITLE
*: Add missing error checks and enable errcheck in static-check

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

The errcheck analysis is disabled because there are still many places where the returned error is unchecked.

### What is changed and how it works?

Add missing error checks and enable errcheck.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pump/storage/log.go [hunk 1]
```diff
@@ -369,7 +369,7 @@ func seekToNextRecord(reader *bufio.Reader) (bytes int, err error) {
 			return
 		}
 
-		reader.Discard(1)
+		reader.Discard(1) //nolint: errcheck
```
Comment (july2993) [inline pump/storage/log.go]: Add comment for why it will never happen.
and just panic or `var _ = reader.Discard(1)` ?

Comment (suzaku) [reply pump/storage/log.go]: Fixed.

**Pull request merged**